web/client/src/library/components/sourceList/SourceList.tsx

```diff
@@ -36,76 +39,187 @@ export default function SourceList<
   to: string
   items?: TItem[]
   types?: TType
+  isActive?: (id: string) => boolean
   byName?: string
   disabled?: boolean
   byDescription?: string
   className?: string
 }): JSX.Element {
   const [filter, setFilter] = useState('')
 
-  const filtered =
-    filter === ''
-      ? items
-      : items.filter(item => {
-          const id = item[by] ?? ''
-          const description = String(
-            isNil(byDescription) ? '' : item?.[byDescription] ?? '',
-          )
-          const name = String(isNil(byName) ? '' : item?.[byName] ?? '')
-          const type = String(types?.[id] ?? '')
-
-          return (
-            name.includes(filter) ||
-            description.includes(filter) ||
-            type.includes(filter)
-          )
-        })
+  const scrollableAreaRef = useRef<HTMLDivElement>(null)
+
+  const getActiveIndex = (itemsList: TItem[]): number =>
+    itemsList.findIndex(item => isNotNil(isActive) && isActive(item[by]))
+
+  const [activeItemIndex, filtered] = useMemo(() => {
+    if (filter === '') {
+      return [getActiveIndex(items), items]
```

Comment:
we can handle this case inside `items.forEach` and remove `getActiveIndex`

Reply:
dang it, thanks!

Reply:
are you thinking that since we have to loop over at least some of the array anyway that we should do the forEach regardless of whether there is a filter? (either way since getActiveIndex is no longer used more than once I can get rid of it)

Reply:
yes, we have to loop over items to get activeIndex anyway. filter is kinda like side effect here